go/vt/mysqlctl/replication.go

```diff
@@ -197,12 +197,33 @@ func (mysqld *Mysqld) SetSuperReadOnly(on bool) error {
 
 // WaitMasterPos lets slaves wait to given replication position
 func (mysqld *Mysqld) WaitMasterPos(ctx context.Context, targetPos mysql.Position) error {
-	// Get a connection.
-	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
-	if err != nil {
-		return err
+	var conn mysql.ReplicationActor
+
+	// First check if filePos flavored Position was passed in. If so, we can't defer to the flavor in the connection,
+	// unless that flavor is also filePos.
+	if targetPos.MatchesFlavor(mysql.FilePosFlavorID) {
+		filePositioner := func(cp *mysql.ConnParams) *mysql.ConnParams {
+			outcp := *cp
+			outcp.Flavor = mysql.FilePosFlavorID
+			return &outcp
+		}
+		params, err := mysqld.dbcfgs.DbaConnector().MysqlParamsWithOptions(filePositioner)
+		if err != nil {
+			return err
+		}
+		conn, err = mysql.Connect(ctx, params)
+		if err != nil {
+			return err
+		}
```

Comment:
Is there a need to `conn.Recycle()`?

Reply:
For `mysql.Connect` there is no `Recycle()`. Recycle is a method available to a connection coming from a connection pool. In this case, we are making a direct connection to override the flavor.

Reply:
It sounds like this code is going away anyway, but for future reference, we would need to `defer conn.Close()` in this case.